lib.php

```diff
@@ -282,7 +282,7 @@ function logstore_xapi_extract_events($limitnum = 0, $log = XAPI_REPORT_SOURCE_L
     global $DB;
 
     $conditions = null;
-    if ($type) {
+    if (!empty($type) || $type == XAPI_IMPORT_TYPE_LIVE) {
```

Comment:
Perhaps we want `===` here?

Reply:
=== is useful to match value and type but is type important when using define to declare a constant?

Reply:
Why don't we just get rid of the optional'ness of the (2nd and 3rd?) params if they are always provided?
That way, you can get rid of the if statement entirely

Reply:
I've just checked, and everywhere the function is used all three params are provided
```
gordonmacqueen@LPOOL55- store/xapi (master-jisc) » grep -r logstore_xapi_extract_events *                                                            141 ↵
classes/task/failed_task.php:        $extractedevents = logstore_xapi_extract_events($batchsize, XAPI_REPORT_SOURCE_LOG, XAPI_IMPORT_TYPE_FAILED);
classes/task/historical_task.php:        $extractedevents = logstore_xapi_extract_events($batchsize, XAPI_REPORT_SOURCE_LOG, XAPI_IMPORT_TYPE_HISTORIC);
classes/task/emit_task.php:        $extractedevents = logstore_xapi_extract_events($batchsize, XAPI_REPORT_SOURCE_LOG, XAPI_IMPORT_TYPE_LIVE);
lib.php:function logstore_xapi_extract_events($limitnum = 0, $log = XAPI_REPORT_SOURCE_LOG, $type = null)
```